scripts/mirror-builtin-actors.sh

```
@@ -0,0 +1,126 @@
+        body=$(curl --silent "$page_url")
+        next_page_url=$(get_next_page_url "$response")
+
+        echo "$body" | jq --compact-output '.[]' | while read -r release; do
+            TAG_NAME=$(echo "$release" | jq --raw-output '.tag_name') || echo "Error: $release, could not get tag name"
+            PUBLISHED_DATE=$(echo "$release" | jq --raw-output '.published_at')
+            PUBLISHED_DATE_SEC=$(date --date="$PUBLISHED_DATE" +%s)
+
+            if echo "$TAG_NAME" | grep --extended-regexp '^v[0-9]+\.[0-9]+\.[0-9]+.*$' && [[ "$PUBLISHED_DATE_SEC" -ge "$THREE_YEARS_AGO" ]]; then
+                mkdir --parents "$BASE_FOLDER/$TAG_NAME"
+            fi
+        done
+
+        page_url="$next_page_url"
+    done
+}
+
+fetch_and_process_releases
+
+declare -a failed_uploads
+failed_uploads=()
+
+: 'Loop through each version directory to process and upload assets'
+while IFS= read -r version_dir; do
+    TAG_NAME=${version_dir#"$BASE_FOLDER"/}
+    VERSION_DIR="$version_dir"
+    if [ -d "$VERSION_DIR" ]; then
+        echo "Entering directory: $VERSION_DIR"
+
+        tag_url="$API_URL/tags/$TAG_NAME"
+        release=$(curl --silent "$tag_url")
+
+        # Check if the assets array is not null
+        if [[ $(echo "$release" | jq '.assets') != "null" ]]; then
+            ASSETS=$(echo "$release" | jq --compact-output '.assets[]')
+
+            pushd "$VERSION_DIR" > /dev/null
+            echo "Processing assets for $TAG_NAME..."
+
+            echo "$ASSETS" | while IFS= read -r asset; do
+                DOWNLOAD_URL=$(echo "$asset" | jq --raw-output '.browser_download_url')
+                FILE_NAME=$(echo "$asset" | jq --raw-output '.name')
+
+                echo "Checking asset: $FILE_NAME"
+                if [ ! -f "$FILE_NAME" ]; then
+                    echo "Downloading $FILE_NAME..."
+                    curl --location --silent --show-error --output "$FILE_NAME" "$DOWNLOAD_URL" || echo "Failed to download $FILE_NAME"
```

This silently continues the script on failure, no?